pyproject.toml

```diff
@@ -249,6 +249,10 @@ filterwarnings = [
     # ignore internal nbmake warnings
     'ignore:unclosed \<socket.socket:ResourceWarning',
     'ignore:unclosed event loop \<:ResourceWarning',
+    # ignore warnings generated while running tests
+    "ignore::DeprecationWarning",
+    "ignore::UserWarning",
+    "ignore::RuntimeWarning",
```

Comment:
We probably shouldn't ignore `UserWarning`s and `RuntimeWarning`s either, does not ignoring them fail the tests?

Reply:
Oh, thanks the clarifications! We can revisit this sometime soon, then.